testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oauth/tokenexchange/ClientTokenExchangeAudienceAndScopesTest.java

```diff
@@ -76,10 +78,17 @@ public void testAudienceRequested() throws Exception {
 
     @Test
     public void testUnavailableAudienceRequested() throws Exception {
-        String accessToken = resourceOwnerLogin("john", "password", List.of("target-client1"), List.of("default-scope1"));;
+        String accessToken = resourceOwnerLogin("john", "password", List.of("target-client1"), List.of("default-scope1"));
+        // request invalid client audience
+        AccessTokenResponse response = oauth.doTokenExchange(accessToken, List.of("target-client1", "invalid-client"), "requester-client", "secret", null);
+        Assert.assertEquals(Response.Status.BAD_REQUEST.getStatusCode(), response.getStatusCode());
+        Assert.assertEquals(OAuthErrorException.INVALID_CLIENT, response.getError());
+        Assert.assertEquals("Audience not found", response.getErrorDescription());
         // The "target-client3" is valid client, but unavailable to the user. Request allowed, but "target-client3" audience will not be available
```

Comment:
Minor: This comment is outdated now (no need to change PR unless there are more other changes needed as this can be fixed in some follow-up)